docarray/index/backends/milvus.py

```diff
@@ -0,0 +1,841 @@
+            default is {'protocol': 'protobuf'}.
+        :param default_column_config: Dictionary that defines the default configuration
+            for each data type column.
+        """
+
+        index_name: Optional[str] = None
+        collection_description: str = ""
+        host: str = "localhost"
+        port: int = 19530
+        user: Optional[str] = ""
+        password: Optional[str] = ""
+        token: Optional[str] = ""
+        consistency_level: str = 'Session'
+        search_params: Dict = field(
+            default_factory=lambda: {
+                "params": {"nprobe": 10},
+            }
+        )
+        serialize_config: Dict = field(default_factory=lambda: {"protocol": "protobuf"})
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: defaultdict(
+                dict,
+                {
+                    DataType.FLOAT_VECTOR: {
+                        'index_type': 'IVF_FLAT',
+                        'metric_type': 'L2',
+                        'params': {"nlist": 1024},
+                    },
+                },
+            )
+        )
+
+    @dataclass
+    class RuntimeConfig(BaseDocIndex.RuntimeConfig):
+        """Dataclass that contains all "dynamic" configurations of RedisDocumentIndex.
+
+        :param batch_size: Batch size for index/get/del.
+        """
+
+        batch_size: int = 100
+
+    class QueryBuilder(BaseDocIndex.QueryBuilder):
+        def __init__(self, query: Optional[List[Tuple[str, Dict]]] = None):
+            super().__init__()
+            # list of tuples (method name, kwargs)
+            self._queries: List[Tuple[str, Dict]] = query or []
+
+        def build(self, *args, **kwargs) -> Any:
+            """Build the query object."""
+            return self._queries
+
+        find = _collect_query_args('find')
+        filter = _collect_query_args('filter')
+        text_search = _raise_not_supported('text_search')
+        find_batched = _raise_not_composable('find_batched')
+        filter_batched = _raise_not_composable('filter_batched')
+        text_search_batched = _raise_not_supported('text_search_batched')
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        """Map python type to database type.
+        Takes any python type and returns the corresponding database column type.
+
+        :param python_type: a python type.
+        :return: the corresponding database column type, or None if ``python_type``
+        is not supported.
+        """
+        type_map = {
+            int: DataType.INT64,
+            float: DataType.FLOAT,
+            str: DataType.VARCHAR,
+            bytes: DataType.VARCHAR,
+            np.ndarray: DataType.FLOAT_VECTOR,
+            list: DataType.FLOAT_VECTOR,
+            AnyTensor: DataType.FLOAT_VECTOR,
+            AbstractTensor: DataType.FLOAT_VECTOR,
+        }
```

Comment:
We can move this dict away from the function imo

Reply:
hmm but all other backends have it inside